src/main/java/io/supertokens/storage/mysql/queries/PasswordlessQueries.java

```diff
@@ -180,10 +180,9 @@ public static void createCode(Start start, PasswordlessCode code)
                 sqlCon.commit();
             } catch (SQLException e) {
 
-                String message = e.getMessage();
-                if (message.contains("foreign key constraint fails")
-                        && message.contains(Config.getConfig(start).getPasswordlessCodesTable())
-                        && message.contains("device_id_hash")) {
+                if (e.getMessage().contains("foreign key constraint fails")
```

Comment:
My issue was not with `message.contains(` part.. but was with your using "foreign key constraint fails" as the matching criteria instead of "foreign key" (as is done in other cases..)

Reply:
done.